src/Stat.php

```diff
@@ -431,26 +431,25 @@ public static function getData($itemtype, $type, $date1, $date2, $start, array $
      **/
     public static function showTable($itemtype, $type, $date1, $date2, $start, array $value, $value2 = "")
     {
-        // Set display type for export if define
-        $output_type = $_GET["display_type"] ?? Search::HTML_OUTPUT;
-        $output = SearchEngine::getOutputForLegacyKey($output_type);
-        $is_html_output = is_a($output, HTMLSearchOutput::class);
-
         $numrows = count($value);
         if ($numrows === 0) {
+            $output = SearchEngine::getOutputForLegacyKey(Search::HTML_OUTPUT);
             echo $output::showHeader(0, 0);
-            if ($is_html_output) {
-                echo '<div class="alert alert-info">' . __s('No statistics are available') . '</div>';
-            } else {
-                echo __('No statistics are available');
-            }
```

Comment:
We shouldn't assume that an export with no results cannot be called. The option to export is removed in the web UI, but direct GET requests, showgraph option in some other stat pages, API calls, etc could potentially result in an export with this HTML alert content.

Reply:
Well... It's the same for all exports, no?

Reply:
At least, the behavior is almost the same: an HTML page with "No statistics available" is displayed. We just have an extra div with my changes - is that really a big issue?

Also, I do not have in mind we can call CSV exports from any API endpoint...

Reply:
The issue is with the alert div potentially showing in non-HTML display types and making their contents invalid.
At the very least, it will be possible to get these exports from the new API.

Reply:
Maybe the more correct option is to show the alert for HTML output, and nothing for the other exports.

Reply:
HTML message is now only displayed when output is HTML. Otherwise, it throws an exception (just a blank page without any information seems not a good idea).